src/main/java/uk/ac/sanger/sccp/stan/service/WorkProgressServiceImp.java

```diff
@@ -192,6 +193,18 @@ public Map<String, LocalDateTime> loadOpTimes(Work work,
         return opTimes;
     }
 
+    /**
+     * Calculates the last operation that occurred
+     * @param workProgressTimestamps the list of workProgressTimestamps (work peration times)
+     */
+    public String getMostRecentOperation(List<WorkProgressTimestamp> workProgressTimestamps) {
+        if (workProgressTimestamps != null && !workProgressTimestamps.isEmpty()) {
```

Comment:
Instead of sorting, something like `workProgressTimestamp.stream().max(Comparator...)`